.pre-commit-config.yaml

```diff
@@ -27,3 +22,17 @@ repos:
   - id: check-yaml
   - id: debug-statements
   - id: check-added-large-files
+  - id: requirements-txt-fixer
+  - id: file-contents-sorter
+    files: CONTRIBUTORS.txt
+# Another entry is required to apply file-contents-sorter to another file
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: 'v3.3.0'
+  hooks:
+  - id: file-contents-sorter
+    files: docs/spelling_wordlist.txt
+- repo: https://github.com/asottile/pyupgrade
+  rev: 'v2.7.3'
+  hooks:
+  - id: pyupgrade
+    args: ['--py36-plus']
```

Comment:
Putting the formatters at the end is usually a bad idea. Because the linters will then check the contents as they were before formatting making it rather inconsistent.

Reply:
Good point.

Reply:
I've pushed flake8 at the last position, all previous hooks eighter modifies python source or don't check them

Reply:
Oh, and another piece of advice: you can split the hooks from the same repo into two and put in different places if needed. Within the blocks of mutating and non-mutating checks, I normally sort the individual hooks so that the fastest ones would go first. This maximizes the responsiveness of the process.

Reply:
One more thing: you can have a separate pr-commit config for very heavy checks and would be nice to always run in CI but users would only invoke them on-demand.

Reply:
Got you